app/models/category_localization.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class CategoryLocalization < ActiveRecord::Base
+  belongs_to :category
+
+  validates :locale, presence: true, length: { maximum: 20 }
+  validates :name, presence: true, length: { maximum: 50 }
+  validates :category_id, uniqueness: { scope: :locale }
+
+  after_commit :invalidate_site_cache
+
+  def invalidate_site_cache
+    Site.clear_cache
```

Comment:
I don't think this is correct. We should only invalidate the cache for the locale that is affected and not the system's locale.

Reply:
It will always clear the cache for the system's default locale unless you call `I18n.with_locale(locale) { Site.clear_cache }`